src/Umbraco.Web.UI.Client/src/common/services/mediahelper.service.js

```diff
@@ -31,7 +31,9 @@ function mediaHelper(umbRequestHelper, $log) {
                     }
                 }
                 else if (mediaEntity.isFolder === undefined) {
-                    mediaEntity.isFolder = mediaEntity.metaData.ContentTypeAlias && mediaEntity.metaData.ContentTypeAlias.toLowerCase() === "folder";
+                    // We should never get here, resolveFileFromEntity should have worked
+                    var alias = mediaEntity.metaData.ContentTypeAlias.toLowerCase();
+                    mediaEntity.isFolder = mediaEntity.metaData.ContentTypeAlias && (alias.endsWith("folder"));
```

Comment:
@nul800sebastiaan This check is weird, there is no point in checking the existence of `mediaEntity.metaData.ContentTypeAlias` since you've already `toLowerCase` it above. If it didn't exist, then an exception would have already been thrown. This null check needs to be fixed.